File: realtor/html/js/localization.js
let Localization = {}

function SetLocales(localization) {
    Localization = localization
}


function GetLocale(locale) {
    if (Localization.languages[Localization.language] && Localization.languages[Localization.language][locale]) {
        return Localization.languages[Localization.language][locale]
    }

    Post({locale:locale}, "AddMissingLocale");
    return locale
}

File: configs/localization.lua
Localization = {
    language = "EN",
    languages = {}
}


local missingLocales = {}

function GetLocale(locale)
    if Localization.languages[Localization.language] and Localization.languages[Localization.language][locale] then
        return Localization.languages[Localization.language][locale]
    else
        missingLocales[locale] = true
        return locale
    end
end


function AddMissingLocale(locale)
    missingLocales[locale] = true
end

function PrintMissingLocales()
    for k, v in pairs(missingLocales) do
        print(string.format('["%s"] = ""', k))
    end
end
